src/core/public/chrome/ui/header/header.tsx

```diff
@@ -281,93 +284,186 @@ export function Header({
+            />
+          </EuiHeaderSectionItem>
+        )}
 
-      <EuiHeaderSectionItem border={useUpdatedHeader ? 'none' : 'left'}>
-        <HeaderNavControls navControls$={observables.navControlsCenter$} />
-      </EuiHeaderSectionItem>
-    </>
-  );
+        {/* Nav controls left */}
 
-  const renderRightControls = () => (
-    <>
-      {useUpdatedHeader && (
-        <EuiHeaderSectionItem border="none">
-          <HeaderNavControls navControls$={observables.navControlsExpandedRight$} />
+        <EuiHeaderSectionItem border={useUpdatedHeader ? 'none' : 'right'}>
+          <HeaderNavControls side="left" navControls$={observables.navControlsLeft$} />
         </EuiHeaderSectionItem>
-      )}
+      </>
+    );
+  };
 
-      {useUpdatedHeader && (
-        <EuiHeaderSectionItem border="none">
-          <HeaderControlsContainer
-            data-test-subj="headerRightControl"
-            controls$={application.currentRightControls$}
-          />
+  const renderCenterControls = () => {
+    let hasCenterControls;
+    let hasNavControlsExpandedCenter;
+    let hasNavControlsCenter;
+
+    application.currentCenterControls$.subscribe((value) => {
+      hasCenterControls = isTruthy(value);
+    });
+    observables.navControlsExpandedCenter$.subscribe((value) => {
+      hasNavControlsExpandedCenter = isTruthy(value);
+    });
+    observables.navControlsCenter$.subscribe((value) => {
+      hasNavControlsCenter = isTruthy(value);
+    });
+
+    if (
+      !hasCenterControls &&
+      !hasNavControlsExpandedCenter &&
+      !hasNavControlsCenter &&
+      useUpdatedHeader
+    ) {
+      return null;
+    }
+
+    return (
+      <>
+        {useUpdatedHeader && (
+          <EuiHeaderSectionItem border="none">
+            <HeaderNavControls navControls$={observables.navControlsExpandedCenter$} />
+          </EuiHeaderSectionItem>
+        )}
+
+        {useUpdatedHeader && (
+          <EuiHeaderSectionItem border="none">
+            <HeaderControlsContainer
+              data-test-subj="headerCenterControl"
+              controls$={application.currentCenterControls$}
+            />
+          </EuiHeaderSectionItem>
+        )}
+
+        <EuiHeaderSectionItem border={useUpdatedHeader ? 'none' : 'left'}>
+          <HeaderNavControls navControls$={observables.navControlsCenter$} />
         </EuiHeaderSectionItem>
-      )}
+      </>
+    );
+  };
+
+  const renderRightControls = () => {
+    let hasNavControlsExpandedCenter;
+    let hasRightControls;
+    let hasNavControlsRight;
+
+    application.currentRightControls$.subscribe((value) => {
+      hasRightControls = isTruthy(value);
+    });
+    observables.navControlsExpandedCenter$.subscribe((value) => {
+      hasNavControlsExpandedCenter = isTruthy(value);
+    });
+    observables.navControlsRight$.subscribe((value) => {
+      hasNavControlsRight = isTruthy(value);
+    });
+
+    if (
+      !hasRightControls &&
+      !hasNavControlsExpandedCenter &&
```

Comment:
curious why does this matter?

Reply:
So based on my observation, even if there's nothing mounted to those header controls mountpoint . It still renders a empty <EuiHeaderSectionItem>, and taking up spaces with no content. That's where the empty spaces are coming from. So adding this check to remove the container when no content

Reply:
To be clear, my question is caring about expandedCenter when rendering right items... Just curious why this matters here whereas it doesn't anywhere else

Reply:
I added here as the condition, since for old UI, we still have this mountpoint and in case anyone is using it(not as far as I know, also Miki documented this is unused), it can still be rendered as expected, instead of returning null. Just aligning with existing code, which made this "expandedCenter" par of overall right control
```

    return (
      <>
        {useUpdatedHeader && (
          <EuiHeaderSectionItem border="none">
            <HeaderNavControls navControls$={observables.navControlsExpandedRight$} />
          </EuiHeaderSectionItem>
        )}

        {useUpdatedHeader && (
          <EuiHeaderSectionItem border="none">
            <HeaderControlsContainer
              data-test-subj="headerRightControl"
              controls$={application.currentRightControls$}
            />
          </EuiHeaderSectionItem>
        )}

        <EuiHeaderSectionItem border={useUpdatedHeader ? 'none' : 'left'}>
          <HeaderNavControls side="right" navControls$={observables.navControlsRight$} />
        </EuiHeaderSectionItem>
      </>
    );
```

Reply:
Sorry, where is expanded center here? I only see expandedRight

Reply:
it's a typo, it should be expandedRight